examples/source_examples/commonpaper/connector.py

```diff
@@ -0,0 +1,113 @@
+# This connector demonstrates how to fetch and sync data from the Common Paper API.
+""" This connector fetches agreement data from the Common Paper API and syncs it to the destination.
+It handles nested data structures, pagination, and maintains sync state using checkpoints."""
+# See the Technical Reference documentation (https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update)
+# and the Best Practices documentation (https://fivetran.com/docs/connectors/connector-sdk/best-practices) for details
+
+# Import required classes from fivetran_connector_sdk
+from fivetran_connector_sdk import Connector, Logging as log, Operations as op
+import requests  # For making HTTP requests to the Common Paper API
+import json      # For JSON data handling and serialization
+import datetime  # For timestamp handling and UTC time operations
+
+# Base URL for the Common Paper API
+API_URL = "https://api.commonpaper.com/v1/agreements"
+
+def get_headers(api_key):
+    """
+    Generate the headers required for Common Paper API authentication.
+    
+    Args:
+        api_key (str): The API key for authentication
+        
+    Returns:
+        dict: Headers dictionary containing Authorization and Accept headers
+    """
+    return {
+        "Authorization": f"Bearer {api_key}",
+        "Accept": "application/json"
+    }
+
+def fetch_agreements(api_key, updated_at):
+    """
+    Fetch agreements from the Common Paper API with updated_at filter.
+    
+    Args:
+        api_key (str): The API key for authentication
+        updated_at (str): ISO format timestamp to filter agreements updated after this time
+        
+    Returns:
+        dict: JSON response containing agreement data
+        
+    Raises:
+        Exception: If the API request fails with non-200 status code
+    """
+    # Format the URL with the filter parameter
+    url = f"{API_URL}?filter[updated_at_gt]={updated_at}"
+    log.fine(f"Fetching agreements from URL: {url}")
+
+    response = requests.get(url, headers=get_headers(api_key))
+    if response.status_code != 200:
+        log.severe(f"Failed to fetch agreements: {response.status_code} - {response.text}")
+        raise Exception(f"API returned {response.status_code}: {response.text}")
```

Comment:
can specific rather than too broad
```suggestion
        raise RuntimeError(f"API returned {response.status_code}: {response.text}")
```